### PR TITLE
DJAN-107 tequila adoption

### PR DESCRIPTION
**Pull request by jbradberry**
Merging together the old `common` and `structure` roles, and splitting off into its own repo, to support installation using `ansible-galaxy`.

### CODE REVIEW

**Comment by dpoirier** [comment]
Here and elsewhere, I'm wondering when we have more than a couple tasks in a row that only run when the same condition is true, if it'd be clearer if we moved those tasks to another file and used a conditional include instead of having them all in main?

File: tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+- name: ensure fail2ban is turned on
+  service: name=fail2ban state=started
+
+- name: configure fail2ban
+  copy: src=jail.local dest=/etc/fail2ban/jail.local owner=root mode=0644
+  notify:
+    - restart fail2ban
+
+- name: add github to the ssh known hosts
+  lineinfile: line="{{ lookup('file', 'github.pub') }}"
```
Comment (dpoirier) [inline tasks/main.yml]: we can use the `known_hosts` module here (Ansible 1.9+)

**Comment by jbradberry** [comment]
Hm, intriguing.  Yes, I think I might like that.  (The conditional includes, that is.)

**Comment by dpoirier** [comment]
Or we could use the new `block` feature in Ansible 2.0 if we're willing to require Ansible 2.0.  (Which wouldn't bother me at all.)

**Comment by jbradberry** [comment]
I went with the conditional include, for now.

**Comment by dpoirier** [comment]
We need to do an "apt-get update" before trying to install any packages.

**Comment by dpoirier** [comment]
Can we add something like this as the first task, to prevent a bad configuration from locking everyone out of the deployed systems?

```
- name: make sure we've defined some dev users
  fail: msg="Please define one or more developers in the `users` variable"
  when: users|length == 0
```

**Comment by jbradberry** [comment]
@poirier r?  I believe I've addressed all of your review comments.

**Comment by dpoirier** [comment]
:shipit:  I haven't tried it, but I assume you have and it looks reasonable.

**Comment by dpoirier** [comment]
P.S. at work I'm @dpoirier.   The poirier ID is my personal account.

**Pull request merged**